### ISSUE TITLE
preprocessing open-NQ task dataset using hf_model.py causes batching error

### ISSUE THREAD

**Issue by JinhaoLei** [google-research/text-to-text-transfer-transformer#375]
Hi, the output from preprocessing open-NQ task has an attribute "answers"(https://github.com/google-research/google-research/blob/d587293a91169310be817fc5798a62074e05b33f/t5_closed_book_qa/t5_cbqa/preprocessors.py#L153), which is a list of candidate answers without fixed length. This makes it can't be batched using tf dataset .batch() API as per this [line](https://github.com/google-research/text-to-text-transfer-transformer/blob/fbbd7b63a93187d198f0929bfd351ab916550db7/t5/models/hf_model.py#L131). The script will throw error because "answers" has different length across different examples. Is there any way to resolve this issue? Thanks!

**Comment by adarob** [comment]
You can remove it from the list of output features since it's only used for eval.

**Comment by adarob** [comment]
@craffel thinks this may be a something we can fix on our end

**Comment by JinhaoLei** [comment]
Yes it would be great if you can fix it as we users do need to have that feature during evaluation, though for training we don't need it.

**Comment by craffel** [comment]
I probably will not get to this for a few weeks. Feel free to hack in a solution where you filter out the non-inputs/targets features during training in the meantime.
